src/opentelemetry/launcher/configuration.py

```diff
@@ -321,11 +302,59 @@ def configure_opentelemetry(
                 "Unable to get hostname, %s", no_hostname_message
             )
 
-    get_tracer_provider().resource = Resource(resource_attributes)
+    tracer_provider = get_tracer_provider()
+
+    if isinstance(tracer_provider, TracerProvider):
+
+        # FIXME: Accessing a private attribute here because resource is no
+        # longer settable since:
+        # https://github.com/open-telemetry/opentelemetry-python/pull/1652
+        # The provider resource can now only be set when the provider is
+        # instantiated, which is too soon for this launcher as it is now.
+        # pylint: disable=protected-access
+        resource = tracer_provider._resource
+
+        if service_name is not None:
+            if "service.name" in resource.attributes.keys():
+                _logger.warning(
+                    "service.name is already defined in the resource "
+                    "attributes, overriding with %s",
+                    service_name,
+                )
+            resource_attributes["service.name"] = service_name
+
+        if service_version is not None:
+            if "service.version" in resource.attributes.keys():
+                _logger.warning(
+                    "service.version is already defined in the resource "
+                    "attributes, overriding with %s",
+                    service_version,
+                )
+            resource_attributes["service.version"] = service_version
+
+        original_attributes = resource.attributes
+        original_attributes.update(resource_attributes)
+
+        get_tracer_provider()._resource = Resource(original_attributes)
```

Comment:
Any reason not to use `merge` here? https://opentelemetry-python.readthedocs.io/en/latest/sdk/resources.html?highlight=resource#opentelemetry.sdk.resources.Resource.merge

Reply:
Yes, in this way it is easier to test, because we can mock `Resource` and check the dictionary that is passed to it.

Reply:
```suggestion
        get_tracer_provider()._resource = resource.merge(Resource(resource_attributes))
```
^^^ you could always check the resource on the tracer_provider after this line no?

Reply:
ah, I see, yes, trying that...

Reply:
Hm, no, it does not work because by that moment `Resource` is a mock object...